homeassistant/components/workday/calendar.py

```diff
@@ -0,0 +1,104 @@
+"""Workday Calendar."""
+
+from __future__ import annotations
+
+from datetime import datetime, timedelta
+
+from holidays import HolidayBase
+
+from homeassistant.components.calendar import CalendarEntity, CalendarEvent
+from homeassistant.const import CONF_NAME
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddConfigEntryEntitiesCallback
+
+from . import WorkdayConfigEntry
+from .const import CONF_EXCLUDES, CONF_OFFSET, CONF_WORKDAYS
+from .entity import BaseWorkdayEntity
+
+CALENDAR_DAYS_AHEAD = 365
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: WorkdayConfigEntry,
+    async_add_entities: AddConfigEntryEntitiesCallback,
+) -> None:
+    """Set up the Holiday Calendar config entry."""
+    days_offset: int = int(entry.options[CONF_OFFSET])
+    excludes: list[str] = entry.options[CONF_EXCLUDES]
+    sensor_name: str = entry.options[CONF_NAME]
+    workdays: list[str] = entry.options[CONF_WORKDAYS]
+    obj_holidays = entry.runtime_data
+
+    async_add_entities(
+        [
+            WorkdayCalendarEntity(
+                obj_holidays,
+                workdays,
+                excludes,
+                days_offset,
+                sensor_name,
+                entry.entry_id,
+            )
+        ],
+    )
+
+
+class WorkdayCalendarEntity(BaseWorkdayEntity, CalendarEntity):
+    """Representation of a Workday Calendar."""
+
+    def __init__(
+        self,
+        obj_holidays: HolidayBase,
+        workdays: list[str],
+        excludes: list[str],
+        days_offset: int,
+        name: str,
+        entry_id: str,
+    ) -> None:
+        """Initialize WorkdayCalendarEntity."""
+        super().__init__(
+            obj_holidays,
+            workdays,
+            excludes,
+            days_offset,
+            name,
+            entry_id,
+        )
+        self._attr_unique_id = f"{entry_id}-calendar"
```

Comment:
unique id is unique per platform per integration, so we don't _really_ need to append `-calendar` to it, unless we plan to add more

Reply:
My own preference but sure, I changed it